projects/maker/index.js

```diff
@@ -36,18 +36,31 @@ async function getJoins(block) {
 
   ilks = ilks.filter((ilk) => ilk.output);
 
-  await Promise.all(ilks.map(async (ilk) => {
+  for (let i = 0; i < ilks.length; i++) {
+    let ilk = ilks[i];
     try {
       let gem = (await sdk.api.abi.call({
         block,
         target: ilk.input.target,
         abi: MakerMCDConstants.gem
       })).output;
+      if (!gem) {
+        // retry in case request fail
+        gem = (await sdk.api.abi.call({
+          block,
+          target: ilk.input.target,
+          abi: MakerMCDConstants.gem
+        })).output;
+      }
+      if (!gem) {
+        throw('no gem')
+      }
+
+      joins.push(ilk.input.target);
+    } catch (e) {
```

Comment:
Since this will bury any errors, what do you think about either logging, or better yet remove the try/catch and let the error bubble up?

Reply:
so the reason for this part is that to ignore contracts that don't have a `gem` field on them that get caught when we are looking for the `joins` which is why we ignore the error as it should only error if that is the case (assuming the request succeeds) I do agree we should add some logging in case of other errors.